lib/ContextModule.js

```diff
@@ -1059,7 +1067,14 @@ module.exports = webpackEmptyAsyncContext;`;
 
 	getSource(sourceString) {
```

Comment:
```suggestion
	getSource(sourceString, compilation) {
```

and call `getSource` with `compilation`.

But you will notice that you don't have a `compilation` in `codeGeneration`, so add that to the `CodeGenerationContext` as **optional** property (optional to not make it a breaking change).

In `Compilation.codeGeneration()` pass the `compilation` property to the context.

`associatedObjectForCache` is optional, so it's fine to omit it when `compilation` is not provided.

Add a TODO comment to `CodeGenerationContext` to make `compilation` required in webpack 6